### ISSUE TITLE
Add ttnn.cumprod basic implementation

### ISSUE THREAD

**Issue by aczajkowskiTT** [tenstorrent/tt-metal#19892]
Building on #19549 keep implementing ttnn.cumprod based on the definition found here:
https://pytorch.org/docs/stable/generated/torch.cumprod.html

In addition to returning the correct shape, the operation should now return the correct data along the given dimension. Consider starting with a few small tensors in the examples from pytorch.org listed above, or an example below, and verify result when calculating the cumulative product on dim=0, dim=1, and dim=2.  When testing, keep in mind the placement of zeroes, as they will simplify the remainder of the cumulative product.

```
x = torch.tensor([
    [[6, 3, 9, 1],
     [5, 2, 8, 4],
     [7, 3, 3, 2]],

    [[1, 7, 4, 6],
     [8, 5, 2, 3],
     [1, 9, 6, 1]]
])
```
Focus on a single core implementation, and avoid large tensors so that the working set fits in L1 cache. After the trivial tensor cumprod operations are working correctly, try tile-sized tensor shapes.
